Snowflake.Data/Core/SFSessionProperty.cs

```diff
@@ -92,29 +109,81 @@ internal static SFSessionProperties parseConnectionString(String connectionStrin
             foreach (string keyVal in propertyEntry)
             {
                 if (keyVal.Length > 0)
-                {
-                    string[] token = keyVal.Split(new string[] { "=" }, StringSplitOptions.None);
-                    if (token.Length == 2)
+                {                    
+                    string[] tokens = keyVal.Split(new string[] { "=" }, StringSplitOptions.None);
+                    if (tokens.Length != 2)
                     {
-                        try
+                        // https://docs.microsoft.com/en-us/dotnet/api/system.data.oledb.oledbconnection.connectionstring
+                        // To include an equal sign (=) in a keyword or value, it must be preceded 
+                        // by another equal sign. For example, in the hypothetical connection 
+                        // string "key==word=value" :
+                        // the keyword is "key=word" and the value is "value".

```

Comment:
Can you add this to README.md?

Reply:
Yes, good idea.
Also adding the new connection property introduced with the new authenticators.